react/src/components/UserProfileSettingModal.tsx

```diff
@@ -22,212 +48,140 @@ const UserProfileSettingModal: React.FC = () => {
-    };
-  }
-
-  const { isOpenUserPrefDialog } = parsedValue;
-
-  const mutationToUpdateUserFullName = useTanMutation({
-    mutationFn: (values: { email: string; full_name: string }) => {
-      return baiClient.update_full_name(values.email, values.full_name);
-    },
-  });
 
-  const mutationToUpdateUserPassword = useTanMutation({
-    mutationFn: (values: {
-      old_password: string;
-      new_password: string;
-      new_password2: string;
-    }) => {
-      return baiClient.update_password(
-        values.old_password,
-        values.new_password,
-        values.new_password2,
-      );
+  const mutationToRemoveTotp = useTanMutation({
+    mutationFn: (email: string) => {
+      return baiClient.remove_totp(email);
     },
   });
 
-  const onSelectAccessKey = (value: string) => {
-    const matchLoggedAccount = _.find(keyPairInfo.keypairs, [
-      'secret_key',
-      value,
-    ]);
-    form.setFieldsValue({
-      secret_key: matchLoggedAccount.secret_key,
-    });
-  };
-
-  const updateFullName = (newFullName: string) => {
-    if (baiClient.full_name !== newFullName) {
-      mutationToUpdateUserFullName.mutate(
-        {
-          full_name: newFullName,
-          email: baiClient.email,
-        },
-        {
-          onSuccess: () => {
-            messageApi.open({
-              type: 'success',
-              content: t('webui.menu.FullnameUpdated'),
-            });
-            dispatchEvent('updateFullName', { newFullName });
-          },
-          onError: (error: any) => {
-            messageApi.open({
-              type: 'error',
-              content: error.message,
-            });
-          },
-        },
-      );
-    }
-  };
-
-  const updatePassword = (
-    oldPassword: string,
-    newPassword: string,
-    newPassword2: string,
-  ) => {
-    if (!oldPassword && !newPassword && !newPassword2) {
-      dispatchEvent('cancel', null);
-      return;
-    }
-    if (!oldPassword) {
-      messageApi.open({
-        type: 'error',
-        content: t('webui.menu.InputOriginalPassword'),
-      });
-      return;
-    }
-    if (!newPassword) {
-      messageApi.open({
-        type: 'error',
-        content: t('webui.menu.InputNewPassword'),
-      });
-      return;
-    }
-    if (newPassword !== newPassword2) {
-      messageApi.open({
-        type: 'error',
-        content: t('webui.menu.NewPasswordMismatch'),
-      });
-      return;
-    }
-    mutationToUpdateUserPassword.mutate(
-      {
-        old_password: oldPassword,
-        new_password: newPassword,
-        new_password2: newPassword2,
-      },
-      {
-        onSuccess: () => {
+  const onSubmit = () => {
+    form.validateFields().then((values) => {
+      userMutations.updateFullName(values.full_name, {
+        onSuccess: (newFullName) => {
           messageApi.open({
             type: 'success',
-            content: t('webui.menu.PasswordUpdated'),
+            content: t('webui.menu.FullnameUpdated'),
```

Comment:
It always comes out even if I don't change the fullname. It seems better to modify this message to appear only when the fullname is changed.